src/components/DateRangeInput/index.tsx

```diff
@@ -0,0 +1,194 @@
+    const {
+        onDateRangeSelect,
+        onCloseButtonClick,
+    } = props;
+
+    const [startDate, setStartDate] = useInputValue('');
+    const [endDate, setEndDate] = useInputValue('');
+
+    const handleSelectButtonClick = React.useCallback(() => {
+        if (startDate && endDate) {
+            onDateRangeSelect(startDate, endDate);
+        }
+    }, [startDate, endDate, onDateRangeSelect]);
+
+    return (
+        <Modal
+            heading="Select date range"
+            className={styles.dateRangeModal}
+            onClose={onCloseButtonClick}
+            footer={(
+                <Button
+                    disabled={!startDate || !endDate}
+                    onClick={handleSelectButtonClick}
+                >
+                    Select
+                </Button>
+            )}
+        >
+            <DateInput
+                className={styles.startDateInput}
+                label="Start date"
+                value={startDate}
+                onChange={setStartDate}
+            />
+            <DateInput
+                className={styles.endDateInput}
+                label="End date"
+                value={endDate}
+                onChange={setEndDate}
+            />
+        </Modal>
+    );
+}
+
+export interface Props<N> {
+    name?: N;
+    value?: DateRangeValue;
+    onChange: (newValue: DateRangeValue | undefined, name?: N) => void;
+}
+
+// TODO: Expose more props (eg: elementRef, inputElementRef, etc)
+function DateRangeInput<N extends string>(props: Props<N>) {
+    const {
+        name,
+        value,
+        onChange,
+    } = props;
+
+    const [showCustomDateRangeModal, setShowCustomDateRangeModal] = React.useState(false);
+
+    const handleCloseButtonClick = React.useCallback(() => {
+        setShowCustomDateRangeModal(false);
+    }, [setShowCustomDateRangeModal]);
+
+    const handleDateRangeSelect = React.useCallback((startDate, endDate) => {
+        setShowCustomDateRangeModal(false);
+        if (onChange) {
+            onChange({
+                key: 'customRange',
+                startDate,
+                endDate,
+            });
+        }
+    }, [setShowCustomDateRangeModal, onChange]);
+
+    const handleSelectInputChange = React.useCallback((newValue) => {
+        if (newValue === 'customRange') {
+            setShowCustomDateRangeModal(true);
+        } else if (onChange) {
+            if (newValue) {
+                onChange({
+                    key: newValue,
+                });
+            } else {
+                onChange(undefined);
+            }
+        }
+    }, [onChange]);
+
+    const displayValueSelector = React.useCallback((d: DateRangeOption) => {
+        if (d.key !== 'customRange') {
+            return d.label;
+        }
+
+        return `${value?.startDate} to ${value?.endDate}`;
+    }, [value]);
+
+    return (
+        <>
+            <SelectInput
+                name={name}
```

Comment:
`name` here is optional but for `SelectInput` it is required value

Reply:
`name` is not an optional field